Frameworks/UIKit/UIViewController.mm

```diff
@@ -1336,28 +1349,23 @@ - (void)_addToTop:(NSNumber*)animatedValue {
 
     if ([self parentViewController] != nil) {
         if ([self modalPresentationStyle] == UIModalPresentationPopover) {
-            if (priv->_presentationController != nil) {
-                [priv->_presentationController release];
-            }
-
-            priv->_presentationController = priv->_popoverPresentationController;
-
             [priv->_popoverPresentationController _prepareForPresentation];
 
             displayPopover = ![self _hidesParent];
 
             if (displayPopover) {
-                priv->_popoverPresentationController->_managesPresentation = TRUE;
+                priv->_popoverPresentationController->_managesPresentation = YES;
```

Comment:
This is slightly improved by the followup commit. I've still opted to keep the teardown of the UIViewController priv variables inside UIViewController.mm (instead of poking at these from UIPopoverPresentationController.mm). The _prepareForPresentation call needs to stay in UIViewController so that the UIPopoverPresentationDelegate is still informed even when there's no actual popover (non-tablet scenarios).